HW-Week2_Day03_Inheritance_and_ClassTypes.iml

```diff
@@ -1,12 +1,79 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<module type="JAVA_MODULE" version="4">
-  <component name="NewModuleRootManager" inherit-compiler-output="true">
-    <exclude-output />
-    <content url="file://$MODULE_DIR$">
-      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
-    </content>
-    <orderEntry type="inheritedJdk" />
-    <orderEntry type="sourceFolder" forTests="false" />
-    <orderEntry type="library" name="KotlinJavaRuntime" level="project" />
-  </component>
-</module>
\ No newline at end of file
+//عملنا كلاس وخليناهopen نورث منو ونعمل  2-properties
+  open class Shape(var shapeName:String,var shapeColor:String){
+// عملنا funوسميناهاinfo وترجع two partmaterهيا shaper name,shape color
+       fun info(): String {
+        return ("$shapeColor $shapeName")
+       }
+    // عملناfunترجع doubleوعملناهاopen عشان نورث منو بطريقة override
+       open fun computerArea():Double{
+           return 0.0
+       }
+
+
+  }
+//عملنا كلاس Triangleيورث منShapeنورث  properties حقتو
+  class Triangle():Shape("Triangle","Red"){
+    //عرفنا متغريين واعطينهم  doubleواعطينهم قيمة
+     var baseNum:Double=10.0
+     var heightNum:Double=20.0
+    //عملنا معادلة  الرياضية حطينهافي متغير
+     var resultTri=(heightNum*baseNum)/2
+
+    //  عملناfunترجع double بطريقة override نورث ويرجع متغير
+     override fun computerArea(): Double {
+         return resultTri
+     }
+
+  }
+
+//عملنا كلاس Rectangleيورث منShapeنورث  properties حقتو
+     class Rectangle:Shape("Rectangle","Blue"){
+    //عرفنا متغريين واعطينهم  doubleواعطينهم قيمة
+         var widthNum:Double=30.0
+         var heightNumber:Double=20.0
+    //عملنا معادلة  الرياضية حطينهافي متغير
+         var resultRec=widthNum*heightNumber
+    //  عملناfunترجع double بطريقة override نورث ويرجع متغير
```

Comment:
what did we inherit ? why is it important and how it affect the rest of your code ?